packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2841 @@
+  /// [MenuStyle.alignment] on the [style] attribute.
+  ///
+  /// Use this for fine adjustments of the menu placement.
+  ///
+  /// Defaults to the [EdgeInsetsDirectional.start] portion of
+  /// [MenuStyle.padding] on the [style] attribute for menus whose parent menu
+  /// (the menu that the button for this menu resides in) is vertical, and the
+  /// [EdgeInsetsDirectional.top] portion of [MenuStyle.padding] on the [style]
+  /// attribute when it is horizontal.
+  final Offset? alignmentOffset;
+
+  /// A list of children containing the menu items that are the contents of the
+  /// menu surrounded by this [MenuAnchor].
+  ///
+  /// {@macro flutter.material.menu_bar.shortcuts_note}
+  final List<Widget> menuChildren;
+
+  /// The widget that this [MenuAnchor] surrounds.
+  ///
+  /// Typically this is a button used to open the menu by calling
+  /// [MenuController.open] on the `controller` passed to the builder.
+  ///
+  /// If not supplied, then the [MenuAnchor] will be the size that its parent
+  /// allocates for it.
+  final MenuAnchorChildBuilder? builder;
+
+  /// The optional child to be passed to the [builder].
+  ///
+  /// Supply this child if there is a portion of the widget tree built in
+  /// [builder] that doesn't depend on the `controller` or `context` supplied to
+  /// the [builder]. It will be more efficient, since Flutter doesn't need to
+  /// rebuild this child when those change.
+  final Widget? child;
+
+  @override
+  State<MenuAnchor> createState() => _MenuAnchorState();
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return menuChildren.map<DiagnosticsNode>((Widget child) => child.toDiagnosticsNode()).toList();
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(FlagProperty('isBar', value: _isBar, ifTrue: 'BAR'));
+    properties.add(FlagProperty('isPositioned', value: _isPositioned, ifTrue: 'POSITIONED'));
+    properties.add(DiagnosticsProperty<FocusNode?>('focusNode', focusNode));
+    properties.add(DiagnosticsProperty<MenuStyle?>('style', style));
+    properties.add(EnumProperty<Clip>('clipBehavior', clipBehavior));
+    properties.add(DiagnosticsProperty<Offset?>('alignmentOffset', alignmentOffset));
+    properties.add(StringProperty('child', child.toString()));
+  }
+}
+
+class _MenuAnchorState extends State<MenuAnchor> {
+  // This is the global key that is used later to determine the bounding rect
+  // for the anchor's region that the CustomSingleChildLayout's delegate
+  // uses to determine where to place the menu on the screen and to avoid the
+  // view's edges.
+  final GlobalKey _anchorKey = GlobalKey(debugLabel: kReleaseMode ? null : 'MenuAnchor');
+  _MenuAnchorState? _parent;
+  bool _childIsOpen = false;
+  Axis get _orientation => widget._isBar ? Axis.horizontal : Axis.vertical;
+  FocusNode? get _focusNode => widget.focusNode;
```

Comment:
This indirection appears to just add a little more complexity when reading the code. Could it be removed?